arangod/Aql/Expression.h

```diff
@@ -303,6 +303,9 @@ class Expression {
     AttributeAccessor* _accessor;
   };
 
+  // we keep the amount of used bytes by the buffer stored in "_data" here.
+  size_t _usedBytesByData = 0;
```

Comment:
@jsteemann I'm aware of buffer size during allocation - therefore persisting it here for later decrease. Is this fine, or can/should we (re-)calculate the amount used by `_data`

Reply:
It's fine as it is.